static/app/components/issues/groupList.tsx

```diff
@@ -142,7 +144,9 @@ class GroupList extends Component<Props, State> {
 
     const endpoint = this.getGroupListEndpoint();
 
-    const parsedQuery = parseSearch((queryParams ?? this.getQueryParams()).query);
+    const parsedQuery = parseSearch(
+      (queryParams ?? this.getQueryParams()).query as string
```

Comment:
Why do you have to cast it like this?

Reply:
When `queryParams` was optional, it would be more likely to end up using what comes out of `this.getQueryParams()` which is guaranteed to be a string.

Since `queryParams` is now mandatory, it checks for the types and it realizes that it could be a Record of something other than a `string` (`Record<string, number | string | string[] | undefined | null>;`)

This is what I get if I remove `as string`.
<img width="618" alt="image" src="https://github.com/getsentry/sentry/assets/44410/35ab7c21-3bab-4ac7-b0e7-eacbf49c52a7">

I could go back to `queryParams` being optional but making it mandatory makes more sense.